include/pybind11/pytypes.h

```diff
@@ -930,11 +930,10 @@ class str : public object {
 private:
     /// Return string representation -- always returns a new reference, even if already a str
     static PyObject *raw_str(PyObject *op) {
-        PyObject *str_value = PyObject_Str(op);
-        if (!str_value) throw error_already_set();
```

Comment:
This one line went missing but is important! (Unless this can never fail?)

Reply:
(Important to add a test for this as well)

Reply:
Nvm. Just noticed this will be done by the code generated by `PYBIND11_OBJECT_CVT` ;-)

Reply:
> This one line went missing but is important! (Unless this can never fail?)

Oops, sorry, that was an accident (manual cherry-picking). I fixed it quick to get that part out of the way. More changes later.